internal/resource/url.go

```diff
@@ -408,6 +409,18 @@ func (f *Fetcher) fetchFromS3(u url.URL, dest s3target, opts FetchOptions) error
 	}
 	cfg := f.AWSConfig.Copy()
 
+	// Determine if the s3 URI is an access point ARN
+	bucket := u.Host
+	key := u.Path
+	if arn.IsARN(u.Opaque) {
```

Comment:
recommend doing "if Opaque" { ... } else { bucket = u.Host
key = u.Path }" so the vars are set once

Reply:
done